### PR TITLE
fix: block tracker stops polling when switching away from the network

### LINKED ISSUE #17040: [Bug]: MM is sending RPC requests to another custom network, aside from the one that's currently selected
### Describe the bug

**Problem**: after adding a custom localhost network, if I change to another network and terminate the ganache server for localhost, I still can see how requests are being send to the localhost custom network together with the requests to the currently selected network.

https://user-images.githubusercontent.com/54408225/209115431-80f25d71-ef80-4ffc-98d3-fa05be22e1a9.mp4

### Steps to reproduce

[investigating]
1. Setup a ganache server with a different port number than the default `ganache port 8546`
2. On MM, go to Add Network
3. Add the new server pointing to localhost 8546
4. Change to another network
5. Terminate ganache server
6. Observe network requests on the background console -- requests to the localhost net are still being sent, alongside with requests to the current selected network

### Error messages or log output

_No response_

### Version

10.24.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

- This duplicate ticket also provides steps to reproduce: https://github.com/MetaMask/metamask-extension/issues/23415

### PR DESCRIPTION
**Pull request by mikesposito**
## **Description**

This PR bumps `@metamask/network-controller` and `@metamask/eth-json-rpc-middleware` by a patch version to fix an issue related to `@metamask/eth-block-tracker`, used to listen to new blocks emitted by networks.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29045?quickstart=1)

## **Related issues**

Fixes: #17040

## **Manual testing steps**

This only applies to views that show a single chain: since now the wallet home shows all networks, requests will be fired regardless of the globally selected network. To test this, the chain filter in the home should be set to a specific chain, instead of all

1. Add a local ganache network 
2. Turn off the local ganache server
3. Observe requests failing when navigating to the home of the wallet (while showing all networks)
4. Filter a single chain which is not the local one (e.g. mainnet)
5. Polling to localhost should stop

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -242,7 +242,7 @@
     "@expo/config-plugins/glob": "^10.3.10",
     "@solana/web3.js/rpc-websockets": "^8.0.1",
     "@metamask/nonce-tracker@npm:^5.0.0": "patch:@metamask/nonce-tracker@npm%3A5.0.0#~/.yarn/patches/@metamask-nonce-tracker-npm-5.0.0-d81478218e.patch",
-    "@metamask/network-controller@npm:^22.0.0": "patch:@metamask/network-controller@npm%3A22.1.0#~/.yarn/patches/@metamask-network-controller-npm-22.1.0-621c281f70.patch",
+    "@metamask/network-controller@npm:^22.0.0": "patch:@metamask/network-controller@npm%3A22.1.1#~/.yarn/patches/@metamask-network-controller-npm-22.1.0-621c281f70.patch",
```
Comment (Gudahtt) [inline package.json:245]: The patch is still based on 22.1.0 🤔

Comment (mikesposito) [reply package.json]: Ops! updated in 58c5bac5538622c497ab0fb0c08d07c20d143b2a

File: .yarn/patches/@metamask-network-controller-npm-22.1.1-09b6510f1e.patch
Comment (mikesposito) [inline .yarn/patches/@metamask-network-controller-npm-22.1.1-09b6510f1e.patch:1]: Missing `PATCH.txt` - on it

**Comment by mikesposito** [comment]
@SocketSecurity ignore npm/reselect@3.0.1
This package version was published [8 years ago](https://socket.dev/npm/package/reselect/overview/3.0.1), and is not being changed in this PR

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Review by mcmire — DISMISSED** [review]
I have confirmed that requests no longer occur via the block tracker for a failing network when switching away.

**Comment by mikesposito** [comment]
Resolved conflicts

**Review by mcmire — DISMISSED** [review]

**Comment by Gudahtt** [comment]
Ah sorry, I pushed a commit to fix the lock file but then realized immediately after that the branch was outdated, so I removed it again. Sorry for that.

The branch needs an update, but in addition to that, there's a bunch of changes in the lockfile that don't seem to be necessary. I'd recommend resetting the lockfile from `main` (`git checkout origin/main yarn.lock && yarn && yarn dedupe`) to get rid of those.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by cryptodev-2s — APPROVED** [review]

**Pull request merged**